runtime.c

```diff
@@ -212,26 +212,53 @@ print_help()
         "AppImage options:\n\n"
         "  --appimage-extract              Extract content from embedded filesystem image\n"
         "  --appimage-help                 Print this help\n"
-        "  --appimage-mount                Mount embedded filesystem image and\n"
-        "                                  print mount point and wait for kill with Ctrl-C\n"
-        "  --appimage-offset               Print byte offset to start of\n"
-        "                                  embedded filesystem image\n"
+        "  --appimage-mount                Mount embedded filesystem image and print\n"
+        "                                  mount point and wait for kill with Ctrl-C\n"
+        "  --appimage-offset               Print byte offset to start of embedded\n"
+        "                                  filesystem image\n"
         "  --appimage-portable-home        Create a portable home folder to use as $HOME\n"
-        "  --appimage-portable-config      Create a portable config folder to use as $XDG_CONFIG_HOME\n"
+        "  --appimage-portable-config      Create a portable config folder to use as\n"
+        "                                  $XDG_CONFIG_HOME\n"
         "  --appimage-signature            Print digital signature embedded in AppImage\n"
         "  --appimage-updateinfo[rmation]  Print update info embedded in AppImage\n"
         "  --appimage-version              Print version of AppImageKit\n"
         "\n"
         "Portable options:\n"
         "\n"
-        "  If you want the AppImage to use a portable $HOME or $XDG_CONFIG_HOME, you can\n"
-        "  use the --appimage-portable options or create the following directories manually:\n"
+        "There is support to have a portable $HOME and $XDG_CONFIG_HOME for a specific\n"
+        "AppImage. To do so, you can either create a directory as follows:\n"
         "\n"
         "  My.AppImage.home will be used as $HOME\n"
         "  My.AppImage.config will be used as $XDG_CONFIG_HOME\n"
+        "\n"
+        "Or you can use --appimage-portable-home to create a portable $HOME directory\n"
+        "or --appimage-portable-config for a portable $XDG_CONFIG_HOME. These options\n"
+        "cannot be used at the same time"
     );
 }
 
+void
+portable_option(const char *arg, const char *appimage_path, const char *name)
+{
+    char option[32];
+    sprintf(option, "appimage-portable-%s", name);
+
+    if (arg && strcmp(arg, option)==0) {
+        char portable_dir[PATH_MAX];
+        char fullpath[PATH_MAX];
+
+        readlink(appimage_path, fullpath, sizeof(fullpath));
```

Comment:
There should be an error check for IO functions like this one.

Reply:
True, I should have checked the returned value. There's another `readlink` call in this file which returned value is not checked